velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -71,23 +71,21 @@ class JsonParseFunction : public exec::VectorFunction {
       const TypePtr& /* outputType */,
       exec::EvalCtx& context,
       VectorPtr& result) const override {
-    // Initilize errors here so that we get the proper exception context.
+    // Initialize errors here so that we get the proper exception context.
     folly::call_once(
         initializeErrors_, [this] { simdjsonErrorsToExceptions(errors_); });
 
     VectorPtr localResult;
 
     // Input can be constant or flat.
-    // TODO(arpitporwal2293) Replace folly::parseJson with a lightweight
-    // validation of JSON syntax that doesn't allocate memory or copy data.
     assert(args.size() > 0);
     const auto& arg = args[0];
     if (arg->isConstantEncoding()) {
       auto value = arg->as<ConstantVector<StringView>>()->valueAt(0);
-      // Ask the parser to copy the input in case `value' is inlined.
-      auto parsed = parser_.parse(value.data(), value.size(), true);
-      if (parsed.error() != simdjson::SUCCESS) {
-        context.setErrors(rows, errors_[parsed.error()]);
+      paddedInput_.resize(value.size() + simdjson::SIMDJSON_PADDING);
```

Comment:
Unrelated change?

Reply:
It's needed, the DOM parser was doing this in their code, and now because we swapped the DOM parser with ondemand one, we need to do the padding ourselves.